### ISSUE TITLE
Add tree visualisation from D-PLACE

### ISSUE THREAD

**Issue by SimonGreenhill** [clld/grambank#12]

**Comment by SimonGreenhill** [comment]
Note: I think we should just do this on glottolog trees at a family level rather than doing what DPLACE does.

**Comment by HedvigS** [comment]
Loving the tree display in the web app @xrotwang ! Not sure when that happened, but I like it!

Can we make it so that the tips are either language-leveld or the glottocode of the GB dialects? Right now I think it's down to the very end dialect tip of each branch, which for example for Japonic makes it look much larger than if it was viewed by language-leveled tips.

**Comment by xrotwang** [comment]
What exactly are we talking about? Can you post a link?

**Comment by HedvigS** [comment]
Sure! Sorry.

At the very bottom of the family pages there's a classification tree from glottolog: [https://grambank.clld.org/familys/abkh1242](https://grambank.clld.org/familys/abkh1242)

**Comment by HedvigS** [comment]
@SimonGreenhill is it a necessity for you that this is in the launch version of the website? If so, we better get a move on on it probably.

**Comment by HedvigS** [comment]
example from dplace website
![Screenshot 2023-03-01 at 12 55 47](https://user-images.githubusercontent.com/5327845/222132657-e0518641-7766-415e-abd0-19ba3b70d7ac.png)

**Comment by xrotwang** [comment]
@johenglisch the feature was implemented for grambank at some point. Here's the relevant commit https://github.com/clld/grambank/commit/eb8ad9615327df5d202a01f0ad62892403f4b739

**Comment by johenglisch** [comment]
Okay, studid question:  What is actually left to do for this?  I went to

https://grambank.clld.org/familys/indo1319?feature=GB020#3/41.11/49.39

and there's a dropbox for choosing a feature, a map, a table, and a tree.  The main difference I can make out is that d-place allows you to show multiple features in the same tree, while Grambank only visualises one.

**Comment by johenglisch** [comment]
>maybe we should add a statement somewhere saying that this exists?

Hm, maybe, but where?

Currently you can just click on any family in the language table (e.g. https://grambank.clld.org/familys/aust1307) and the family page will prominently show a box at the top.  I suppose we could another box like that to  the page for an individual feature maybe? (like, right next to the *feature combination* box)

**Comment by HedvigS** [comment]
> Okay, studid question: What is actually left to do for this? I went to
> 
> https://grambank.clld.org/familys/indo1319?feature=GB020#3/41.11/49.39
> 
> and there's a dropbox for choosing a feature, a map, a table, and a tree. The main difference I can make out is that d-place allows you to show multiple features in the same tree, while Grambank only visualises one.

wait... how did you get there :)?! I can see the URL, but I haven't yet figured out how to click my way there.

**Comment by xrotwang** [comment]
There used to be a top-level menu entry linking to the list of covered Glottolog families:
https://grambank.clld.org/familys

**Comment by xrotwang** [comment]
> What's the CLLD way of giving an overview of web page functionalities in general?

The "CLLD way" is sticking to uniformity. The main menu links to lists of core objects in the database. The context menu provides relevant links depending on the visited page.

**Comment by HedvigS** [comment]
> There used to be a top-level menu entry linking to the list of covered Glottolog families: https://grambank.clld.org/familys

Okay. Is the idea to insert that again then?

**Comment by HedvigS** [comment]
Okay. Thank you for clarifying that, I've actually been a bit confused for a while what the two menus are about.

Where would you like to put a tab that goes to a page that gives a basic overview of the website functions, like the tree, map, combination of features etc?

**Comment by xrotwang** [comment]
> Where would you like to put a tab that goes to a page that gives a basic overview of the website functions, like the tree, map, combination of features etc?

I'd put the content into a template `help.mako`, which will then picked up by `clld` and automatically linked from an entry "Help" in the context menu of the landing page. @johenglisch see https://github.com/clld/clld/blob/837351371d150f43c4945140d3ec1f2027375f6a/src/clld/web/app.py#L621

**Comment by johenglisch** [comment]
>how to click my way there

 * Go to grambank
 * Click on *Languages and dialects*
 * Click on any family in the language table
 * The family page has a box where you can choose a feature and hit *submit*

This should also work from any place where a language family is clickable (value tables for example).

As for a help page:  Yeah, I can set that up the template.

**Comment by HedvigS** [comment]
> > how to click my way there
> 
> * Go to grambank
> * Click on _Languages and dialects_
> * Click on any family in the language table
> * The family page has a box where you can choose a feature and hit _submit_
> 
> This should also work from any place where a language family is clickable (value tables for example).
> 
> As for a help page: Yeah, I can set that up the template.

Aha okay, I hadn't spotted that the content in the family column were links at all.

**Comment by HedvigS** [comment]
Can we put the tree functionality on the feature page view as well?

**Comment by SimonGreenhill** [comment]
Looks great, thanks @johenglisch. 

* can we check the character mapping e.g. [here](https://grambank.clld.org/familys/aust1307?feature=GB024#4/-4.57/136.85) the map has two states, but the tree has three?
* can we make the colors/symbols match with those on the map? 

Agree with @HedvigS, can we get it on the feature view page. Perhaps we could have the Map and the Tree in alternate tabs/iframes to save on js load/render (i.e. "show me a map, show me a tree")?

**Comment by johenglisch** [comment]
>the map has two states, but the tree has three

Not sure what you mean: From what I can tell both the map and the tree show three values (four if you include `?`).

![states](https://user-images.githubusercontent.com/1862047/223684698-3718c5ec-2d38-495f-bbfb-1a5604441f0e.png)

>can we make the colors/symbols match with those on the map?

Yeah, good idea.  Also, the colours probably shouldn't be orangey red and reddy orange anymore (maybe these: https://github.com/clld/grambank/issues/42#issuecomment-1450015168).

>Perhaps we could have the Map and the Tree in alternate tabs/iframes to save on
>js load/render

I actually had a simpler solution in mind: A drop box on the feature page that lets you pick a language family and brings you to the page with the tree (maybe right next to the one linking to feature combinations; cf. https://github.com/clld/grambank/pull/44#issue-1611226589).

Then we would have a family page that lets you choose a feature and a feature page that lets you choose a family and they would link to the same tree view.  Nice and symmetrical.

**Comment by xrotwang** [comment]
> Then we would have a family page that lets you choose a feature and a feature page that lets you choose a family and they would link to the same tree view. Nice and symmetrical.

Yes, and this page would also be bookmarkable - without relying on URL fragments, etc.

**Comment by HedvigS** [comment]
Looks like this is wrapped up, yeah?

**Comment by HedvigS** [comment]
@SimonGreenhill can you have a look at #49 ?
